src/lib/appContainer.ts

```diff
@@ -13,83 +14,93 @@ import {
   IGovNotifyGateway,
 } from "../gateways/govNotifyApiGateway";
 import {
-  AuthenticateUser,
-  IAuthenticateUser,
+  authenticateUser,
+  AuthenticateUserFn,
 } from "../useCases/authenticateUser";
 import {
-  CreateRegistration,
-  ICreateRegistration,
-} from "../useCases/createRegistration";
-import { IRedirectUserTo, RedirectUserTo } from "../useCases/redirectUserTo";
+  clearCachedRegistration,
+  ClearCachedRegistration,
+} from "../useCases/clearCachedRegistration";
+import { getAccessToken, GetAccessTokenFn } from "../useCases/getAccessToken";
 import {
-  ISendGovNotifyEmail,
-  SendGovNotifyEmail,
-} from "../useCases/sendGovNotifyEmail";
+  getCachedRegistration,
+  GetCachedRegistrationFn,
+} from "../useCases/getCachedRegistration";
+import { redirectUserTo, RedirectUserToFn } from "../useCases/redirectUserTo";
 import {
-  IVerifyFormSubmissionCookieIsSet,
-  VerifyFormSubmissionCookieIsSet,
+  retrieveUserFormSubmissionId,
+  UserFormSubmissionIdFn,
+} from "../useCases/retrieveUserFormSubmissionId";
+import {
+  sendConfirmationEmail,
+  SendConfirmationEmailFn,
+} from "../useCases/sendConfirmationEmail";
+import {
+  submitRegistration,
+  SubmitRegistrationFn,
+} from "../useCases/submitRegistration";
+import {
+  verifyFormSubmissionCookieIsSet,
+  VerifyFormSubmissionCookieIsSetFn,
 } from "../useCases/verifyFormSubmissionCookieIsSet";
 
 export interface IAppContainer {
-  getAuthenticateUser: () => IAuthenticateUser;
-  getCreateRegistration: () => ICreateRegistration;
-  getSendGovNotifyEmail: () => ISendGovNotifyEmail;
-  getVerifyFormSubmissionCookieIsSet: () => IVerifyFormSubmissionCookieIsSet;
-  getRedirectTo: () => IRedirectUserTo;
+  /* Use cases */
+  authenticateUser: AuthenticateUserFn;
+  submitRegistration: SubmitRegistrationFn;
+  sendConfirmationEmail: SendConfirmationEmailFn;
+  verifyFormSubmissionCookieIsSet: VerifyFormSubmissionCookieIsSetFn;
+  redirectUserTo: RedirectUserToFn;
+  userFormSubmissionId: UserFormSubmissionIdFn;
```

Comment:
As discussed, would remove pure functions from this container and just call them directly in getServerSideProps